lib/compiler.js

```diff
@@ -288,63 +289,63 @@ function getFileDependencies (compiler) {
 }
 
 /**
- * @type {WeakMap<WebpackCompilation, WeakMap<HtmlWebpackChildCompiler, boolean>>}}
+ * @type {WeakMap<object, Snapshot>}
  */
-const hasOutdatedCompilationDependenciesMap = new WeakMap();
+const snapshotMap = new WeakMap();
+const fileDepependenciesKeyRef = {};
```

Comment:
This looks really dangerous as it is one single global variable used for (potentially) multiple compilers, compilations and html-webpack-plugin instances.

Reply:
Use a weakmap? or a weakmap on compiler?